From c12d3b95ac6ecfb76069063623b44013f63019f5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Aur=C3=A9lien=20COUDERC?= <coucouf@coucouf.fr>
Date: Mon, 10 Feb 2025 23:38:44 +0100
Subject: [PATCH 1/4] Revert "autotests/integration: make outputchanges test
 initial parsing failure with multiple outputs"

This reverts commit cdcfaf09f7a1341d5feca8193619456de0250b9e.
---
 autotests/integration/outputchanges_test.cpp | 32 ++------------------
 1 file changed, 2 insertions(+), 30 deletions(-)

diff --git a/autotests/integration/outputchanges_test.cpp b/autotests/integration/outputchanges_test.cpp
index c9a0ff9968..3e582070f0 100644
--- a/autotests/integration/outputchanges_test.cpp
+++ b/autotests/integration/outputchanges_test.cpp
@@ -1798,8 +1798,6 @@ void OutputChangesTest::testSettingRestoration_initialParsingFailure()
     const auto edid = file.readAll();
 
     // first, libdisplay-info failed to parse the EDID and we don't have an EDID ID
-    // note that this uses two displays with the same EDID,
-    // to additionally test the case when EDID ID isn't unique when this happens
     Test::setOutputConfig({
         Test::OutputInfo{
             .geometry = QRect(0, 0, 1280, 1024),
@@ -1812,20 +1810,7 @@ void OutputChangesTest::testSettingRestoration_initialParsingFailure()
             .edid = edid,
             .edidIdentifierOverride = QByteArray(),
             .connectorName = std::nullopt,
-            .mstPath = QByteArrayLiteral("MST-1-1"),
-        },
-        Test::OutputInfo{
-            .geometry = QRect(0, 0, 1280, 1024),
-            .internal = false,
-            .physicalSizeInMM = QSize(598, 336),
-            .modes = {
-                ModeInfo(QSize(1280, 1024), 60000, OutputMode::Flag::Preferred),
-                ModeInfo(QSize(640, 480), 60000, OutputMode::Flags{}),
-            },
-            .edid = edid,
-            .edidIdentifierOverride = QByteArray(),
-            .connectorName = std::nullopt,
-            .mstPath = QByteArrayLiteral("MST-1-2"),
+            .mstPath = std::nullopt,
         },
     });
 
@@ -1874,20 +1859,7 @@ void OutputChangesTest::testSettingRestoration_initialParsingFailure()
             .edid = edid,
             .edidIdentifierOverride = std::nullopt,
             .connectorName = std::nullopt,
-            .mstPath = QByteArrayLiteral("MST-1-1"),
-        },
-        Test::OutputInfo{
-            .geometry = QRect(0, 0, 1280, 1024),
-            .internal = false,
-            .physicalSizeInMM = QSize(598, 336),
-            .modes = {
-                ModeInfo(QSize(1280, 1024), 60000, OutputMode::Flag::Preferred),
-                ModeInfo(QSize(640, 480), 60000, OutputMode::Flags{}),
-            },
-            .edid = edid,
-            .edidIdentifierOverride = std::nullopt,
-            .connectorName = std::nullopt,
-            .mstPath = QByteArrayLiteral("MST-1-2"),
+            .mstPath = std::nullopt,
         },
     });
     outputs = kwinApp()->outputBackend()->outputs();
-- 
2.47.2

